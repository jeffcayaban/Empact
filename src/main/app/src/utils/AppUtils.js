import {API_BASE_URL, AUTHORIZATION_HEADER_FIELD} from '../constants';
import {ACCESS_TOKEN, AUTHORIZATION_TOKEN_TYPE, PETITION_LIST_SIZE} from "../constants";

/** Creates a request object and sends the request to the server. **/
const request = (options) => {
    const headers = new Headers({ 'Content-Type': 'application/json' });
    const accessToken = localStorage.getItem(ACCESS_TOKEN);

    if (accessToken !== undefined && accessToken !== null) {
        headers.append(AUTHORIZATION_HEADER_FIELD, `${AUTHORIZATION_TOKEN_TYPE} ${accessToken}`)
    }

    options = {...options, headers: headers};

    return fetch(options.url, options).then(response =>
            (response.json()).then(jsonResponse => (!response.ok) ? Promise.reject(jsonResponse) : jsonResponse)
    );
};

// ---------------------------- APIs ----------------------------

/** Gets all the petitions **/
export function getAllPetitions(page, size, sort, showClosed) {
    page = page || 0;
    size = size || PETITION_LIST_SIZE;

    return request({
        url: API_BASE_URL + "/petitions?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Gets a single petition given an ID **/
export function getPetition(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/" + petitionId,
        method: 'GET'
    });
}

/** Creates a new petition given its details **/
export function createPetition(petitionData) {
    return request({
        url: API_BASE_URL + "/petitions/create",
        method: 'POST',
        body: JSON.stringify(petitionData)
    });
}

/** Deletes an existing petition given its details **/
export function deletePetition(deletePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/delete",
        method: 'DELETE',
        body: JSON.stringify(deletePetitionData)
    });
}

/** Updates an existing petition with the given details **/
export function editPetition(editPetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/edit/",
        method: 'POST',
        body: JSON.stringify(editPetitionData)
    });
}

/** Logs in into a user account **/
export function login(loginRequest) {
    return request({
        url: API_BASE_URL + "/auth/signInUser",
        method: 'POST',
        body: JSON.stringify(loginRequest)
    });
}

/** Creates a new user account given the account details **/
export function signup(signupRequest) {
    return request({
        url: API_BASE_URL + "/auth/signUpUser",
        method: 'POST',
        body: JSON.stringify(signupRequest)
    });
}

/** Checks the username's availability given a username **/
export function checkUsernameAvailability(username) {
    return request({
        url: API_BASE_URL + "/user/checkUsernameAvailability?username=" + username,
        method: 'GET'
    });
}

/** Deletes a user **/
export function deleteUser(deleteUserRequest) {
    return request({
        url: API_BASE_URL + "/auth/deleteUser",
        method: 'DELETE',
        body: JSON.stringify(deleteUserRequest)
    })
}

/** Fetches the current user **/
export function getCurrentUser() {
    if(!localStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }

    return request({
        url: API_BASE_URL + "/user/getCurrentUser",
        method: 'GET'
    });
}

/** Fetches a user profile given a user's username **/
export function getUserProfile(username) {
    return request({
        url: API_BASE_URL + "/user/getUserProfile?username=" + username,
        method: 'GET'
    });
}

/** Fetches the petitions for a given user **/
export function getUserPetitions(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/user?username=" + username + "&page=" + page + "&size=" + size + "&sort=" + sort
        + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Fetches the most discussed petitions for a given user **/
export function getMostDiscussedPetitionsByUser(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitionsByUser?username=" + username + "&page=" + page +
            "&size=" + size + "&sort=" + sort
            + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Closes a petition given its details **/
export function closePetition(closePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/close",
        method: 'POST',
        body: JSON.stringify(closePetitionData)
    })
}

/** Creates a new argument by expert opinion given the new details **/
export function createArgumentByExpertOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Creates a new argument by popular opinion given the new details **/
export function createArgumentByPopularOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Fetches the arguments for a given petition **/
export function getArgumentsByRootPetitionId(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByRootPetitionId?rootPetitionId=" + petitionId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the arguments for a given argument **/
export function getArgumentsByArgumentId(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByArgumentId?argumentId=" + argumentId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the argument for a given ID **/
export function getArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/" + argumentId,
        method: 'GET'
    });
}

/** Fetches the parent argument for a given argument **/
export function getParentArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/getParentArgumentById?argumentId=" + argumentId,
        method: 'GET',
    });
}

/** Checks whether a petition is closed or not **/
export function isPetitionClosed(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/isPetitionClosed/" + petitionId,
        method: 'GET'
    })
}

/** Fetches the arguments for a given user **/
export function getUserArguments(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByUsernameAndAnonymity?username=" + username +
            "&showAnonymousContent=" + showAnonymousContent + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET'
    })
}

/** Fetches the number of supporting or opposing arguments for a given petition **/
export function countArgsByPetitionIdAndIsSupporting(petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countArgsByPetitionIdAndIsSupporting?petitionId=" + petitionId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Fetches the number of supporting or opposing arguments for a given argument **/
export function countArgsByArgumentIdAndIsSupporting(argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countAllByArgumentIdAndIsSupporting?argumentId=" + argumentId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Deletes an argument given its details **/
export function deleteArgument(deleteArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/delete",
        method: 'DELETE',
        body: JSON.stringify(deleteArgumentData)
    });
}

/** Updates an existing argument by popular opinion with new details **/
export function editArgumentByPopularOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Updates an existing argument by expert opinion with new details **/
export function editArgumentByExpertOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Fetches the most discussed petitions **/
export function getMostDiscussedPetitions(page, size, sort, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitions?page=" + page + "&size=" + size + "&sort=" + sort +
            "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments **/
export function getMostDiscussedArguments(pageSize, showClosed) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArguments?size=" + pageSize + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given petition **/
export function getMostDiscussedArgumentsForPetition(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForPetition?page=" + page + "&size=" + size + "&sort=" + sort +
            "&petitionId=" + petitionId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given argument **/
export function getMostDiscussedArgumentsForArgument(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForArgument?page=" + page + "&size=" + size + "&sort=" + sort +
            "&argumentId=" + argumentId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given user **/
export function getMostDiscussedArgumentsForUser(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForUser?page=" + page + "&size=" + size + "&sort=" + sort +
            "&username=" + username + "&showAnonymousContent=" + showAnonymousContent,
        method: 'GET'
    });
}

/** Fetches the data nodes required for the data visualisation of arguments **/
export function getDataNodes(contentId, isPetition) {
    return request({
        url: API_BASE_URL + "/arguments/getDataNodes?contentId=" + contentId + "&isPetition=" + isPetition,
        method: 'GET'
    });
}

/** Fetches data required for the petition report **/
export function getPetitionReport(petitionId) {
    return request({
        url: API_BASE_URL + "/arguments/getPetitionReport?petitionId=" + petitionId,
        method: 'GET'
    });
}

/** Searches for petitions given a query **/
export function searchPetitions(page, size, sort, showClosed, query) {
    return request({
        url: API_BASE_URL + "/petitions/search?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed
         + "&query=" + query,
        method: 'GET'
    });
}
