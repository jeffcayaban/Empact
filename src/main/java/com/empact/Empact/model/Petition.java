package com.empact.Empact.model;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

/**
 * Stores the contents of a user argument.
 */

@Data
@Entity
@Table(name = "petitions")
public class Petition extends BaseContent {

	/** The ID of the petition. **/
	private @Id @GeneratedValue(strategy = GenerationType.IDENTITY) Long id;

	// Contents of a Petition

	/** The title of the petition. **/
	private @NotBlank @Size(max = 90) String title;

	/** The situation of the petition. **/
	private @NotBlank @Size(max = 3000) String situation;

	/** The proposed action of the petition. **/
	private @NotBlank @Size(max = 3000) String action;

	/** The proposed action of the petition. **/
	private @NotBlank @Size(max = 3000) String goal;

	/** The proposed action of the petition. **/
	private @NotBlank @Size(max = 3000) String value;

	/** The closing  of the petition. **/
	private @NotNull Instant closingDateTime;

	/** The list of arguments that are made towards it. **/
	@OneToMany(mappedBy = "rootPetition", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
	private List<Argument> arguments = new ArrayList<>();
}
