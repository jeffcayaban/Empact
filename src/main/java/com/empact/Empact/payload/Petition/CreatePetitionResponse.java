package com.empact.Empact.payload.Petition;

import lombok.AllArgsConstructor;
import lombok.Data;

/**
 * The response object containing the data that denotes the result of the petition creation operation.
 */

@Data
@AllArgsConstructor
public class CreatePetitionResponse {

	/** Indicates whether the operation was successful. **/
	private Boolean success;

	/** A message describing the outcome of the operation. **/
	private String message;

	/** The ID of the newly created petition. **/
	private Long petitionId;
}
