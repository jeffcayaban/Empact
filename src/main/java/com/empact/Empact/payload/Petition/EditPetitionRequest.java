package com.empact.Empact.payload.Petition;

import lombok.Data;

import javax.validation.constraints.NotNull;

/**
 * The request object containing the new data to applied to an existing petition.
 */

@Data
public class EditPetitionRequest extends PetitionRequest {

	/** The ID of the petition to be modified. **/
	private @NotNull Long petitionId;
}
