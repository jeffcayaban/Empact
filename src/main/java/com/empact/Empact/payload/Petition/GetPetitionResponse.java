package com.empact.Empact.payload.Petition;

import com.empact.Empact.payload.User.UserSummary;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

/**
 * The response object containing the data of a given petition.
 */

@Data
@AllArgsConstructor
@NoArgsConstructor
public class GetPetitionResponse {

	/** The ID of the petition. **/
	private Long id;

	/** The title of the petition. **/
	private String title;

	/** The situation of the petition. **/
	private String situation;

	/** The proposed action of the petition. **/
	private String action;

	/** The proposed action of the petition. **/
	private String goal;

	/** The proposed action of the petition. **/
	private String value;

	/** The timestamp of when the petition is created. **/
	private Instant creationDateTime;

	/** The timestamp of when the petition is last updated. **/
	private Instant lastUpdatedDateTime;

	/** The user who created the petition. **/
	private UserSummary createdBy;

	/** The closing date of the petition. **/
	private Instant closingDateTime;

	/** Indicates whether the petition was created with anonymity. **/
	private Boolean isAnonymous;

	/** Indicates whether the petition is winning. **/
	private Boolean isWinning;
}
