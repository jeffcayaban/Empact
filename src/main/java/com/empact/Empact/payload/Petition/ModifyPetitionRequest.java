package com.empact.Empact.payload.Petition;

import lombok.Data;

import javax.validation.constraints.NotNull;

/**
 * The request object containing the ID of the petition to be modified.
 */

@Data
public class ModifyPetitionRequest {

	/** The ID of the petition to be modified. **/
	private @NotNull Long petitionId;
}
