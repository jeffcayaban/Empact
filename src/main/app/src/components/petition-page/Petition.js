import React from 'react'

import {withRouter} from 'react-router-dom'
import Loader from "../Loader";
import {
    deletePetition,
    getPetition,
    countArgsByPetitionIdAndIsSupporting
} from "../../utils/AppUtils";

import PetitionContent from "./PetitionContent";
import PetitionActionPanel from "./PetitionActionPanel";
import ContentRetrievalAlert from "../common/alerts/ContentRetrievalAlert";
import {Col, Grid, Row} from "react-bootstrap";

import {formatTimestampToDate, isDatePassed} from "../../utils/Helpers";
import ArgumentVisualisation from "../common/argument-visualisation/ArgumentVisualisation";
import {PETITION} from "../../constants";
import ArgumentPanel from "../common/ArgumentPanel";
import LastUpdated from "../common/LastUpdated";
import CreateArgumentPanel from "./CreateArgumentPanel";
import CreatedByOn from "../common/CreatedByOn";
import ClosedPetitionActions from "./ClosedPetitionActions";

/**
 * Represents the page that will display the details of the petition. It will additionally display any arguments that
 * support or oppose the petition.
 */

class Petition extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            isLoading: true,
            petition: {},
            isPetitionCreator: false,
            showCreateArgument: true
        };

        this.hideCreateArgumentInvite = this.hideCreateArgumentInvite.bind(this);
    }

    // Fetches the petition from the server.
    getPetitionFromServer(petitionId) {
        getPetition(petitionId)
            .then(response => {
                this.setState({
                    isLoading: false,
                    petition: response,
                    isPetitionCreator: this.props.currentUser && response.createdBy.username === this.props.currentUser.username
                });
            }).catch(error => this.setState({ isLoading: false }));
    }

    // Fetches the number of arguments that supports and opposes the petition.
    getNoOfArguments(petitionId, isSupporting) {
        countArgsByPetitionIdAndIsSupporting(petitionId, isSupporting)
            .then(response => {
                if (isSupporting) {
                    // Assumes the response is for supporting arguments.
                    this.setState({ noOfForArguments: response })
                } else {
                    // Assumes the response is for opposing arguments.
                    this.setState({ noOfAgainstArguments: response })
                }
            }).catch(error => console.error(error))
    }

    // Sends the request to delete the petition.
    deletePetition() {
        return deletePetition({ petitionId: this.state.petition.id })
    }

    // Updates the component's state to hide the create argument invite.
    hideCreateArgumentInvite() {
        this.setState({ showCreateArgument: false })
    }

    // Generates the panel that contains the actions that can be applied to the petition.
    generatePetitionActionPanel(petition) {
        if (!isDatePassed(petition.closingDateTime)) {
            // Assumes that the petition is open.
            return (
                <PetitionActionPanel
                    petition={petition}
                    deletePetition={this.deletePetition.bind(this)}
                    {...this.props}
                />
            )
        } else {
            // Assumes that the petition is closed.
            return <ClosedPetitionActions petition={petition} deletePetition={this.deletePetition.bind(this)} />
        }
    }

    componentDidMount() {
        const petitionId = this.props.match.params.id;
        this.getPetitionFromServer(petitionId);

        // Get Argument Counts
        this.getNoOfArguments(petitionId, true);
        this.getNoOfArguments(petitionId, false);
    }

    render() {
        if (!this.state.isLoading) {
            // Assumes that the data fetching is completed.

            if (Object.keys(this.state.petition).length !== 0) {
                // Assumes that the petition details is defined.

                const { petition, noOfAgainstArguments, isPetitionCreator, noOfForArguments } = this.state;
                const { id, lastUpdatedDateTime, creationDateTime } = petition;
                const ukFormattedCreationDate = formatTimestampToDate(creationDateTime);

                return (
                    <div className={'secondaryPageContainerStyle'}>
                        <Grid className={"divNoPadding"} style={{ paddingTop: '15px' }}>
                            <Col md={12} lg={12} className={'noSidePadding'}>
                                <h1 align="center">{ petition.title }</h1>
                                <div className={"centerStyle"} style={{ paddingTop: '10px' }}>
                                    <CreatedByOn content={petition} date={ukFormattedCreationDate}/>
                                    <LastUpdated
                                        creationDateTime={creationDateTime}
                                        lastUpdatedDateTime={lastUpdatedDateTime}
                                    />
                                </div>
                            </Col>
                        </Grid>
                        <Grid className={"divNoPadding"} style={{ paddingTop: '40px' }}>
                            <Col md={12} lg={12} className={'noSidePadding'}>
                                <div className={"spaceBetweenElements"} style={{ paddingBottom: '15px' }}>
                                    <div className={"contentIdNumber"}>{`ID: ${id}`}</div>
                                    { isPetitionCreator && this.generatePetitionActionPanel(petition) }
                                </div>
                                <PetitionContent petition={petition} showCreatedByOnFooter={false} />
                            </Col>
                        </Grid>
                        <CreateArgumentPanel
                            closingDateTime={petition.closingDateTime}
                            petitionId={id}
                            isPetitionCreator={this.state.isPetitionCreator}
                            hideCreateArgumentInvite={this.hideCreateArgumentInvite}
                            showCreateArgument={this.state.showCreateArgument}
                        />
                        <Grid className={"divNoPadding"} style={{ paddingTop: '30px' }}>
                            <Col md={12} lg={12} className={'noSidePadding'}>
                                <ArgumentVisualisation contentId={id} isPetition={true} />
                            </Col>
                        </Grid>
                        <Grid className={"divNoPadding"}>
                            <Row>
                                <Col md={6} lg={6}>
                                    <ArgumentPanel
                                        isSupporting={true}
                                        subjectId={id}
                                        parentCreatedById={petition.createdBy.id}
                                        noOfArguments={noOfForArguments}
                                        isCreator={isPetitionCreator}
                                        parentContentType={PETITION}
                                    />
                                </Col>
                                <Col md={6} lg={6}>
                                    <ArgumentPanel
                                        isSupporting={false}
                                        subjectId={id}
                                        parentCreatedById={petition.createdBy.id}
                                        noOfArguments={noOfAgainstArguments}
                                        isCreator={isPetitionCreator}
                                        parentContentType={PETITION}
                                    />
                                </Col>
                            </Row>
                        </Grid>
                    </div>
                );
            } else {
                // Assumes that the petition details is not defined, so an alert is displayed.
                return <div className={"alertContainer"}><ContentRetrievalAlert contentType={PETITION} /></div>;
            }
        } else {
            // Assumes that the data is being fetched, so a loader is displayed.
            return <Loader isLoading={this.state.isLoading} />;
        }
    }
}

export default withRouter(Petition);