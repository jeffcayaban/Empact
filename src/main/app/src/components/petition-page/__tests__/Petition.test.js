import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import * as AppUtils from '../../../utils/AppUtils';
import MockPetition from '../../../other/mock-data/MockGetPetitionsResponse';

import Petition from "../Petition";

function createPetitionPage(props = {}) {
    const mockRouter = jest.fn();
    return <Petition.WrappedComponent {...props} params={{ router: mockRouter }} />;
}

describe('Petition', () => {

    it('should render successfully', () => {
        const mockPetitionData = MockPetition.content[0];

        const originalGetPetition = AppUtils.getPetition;
        AppUtils.getPetition = (id) => Promise.resolve(mockPetitionData);

        const originalCountArgs = AppUtils.countArgsByPetitionIdAndIsSupporting;
        AppUtils.countArgsByPetitionIdAndIsSupporting = (_) => Promise.resolve(5);

        const props = {
            match: { params: { id: "testId" }},
            currentUser: { createdBy: { username: 'admin' }}
        };

        const component = shallow(createPetitionPage(props));
        component.setState({ isLoading: false, petition: mockPetitionData });

        expect(toJson(component)).toMatchSnapshot();
        AppUtils.getPetition = originalGetPetition;
        AppUtils.countArgsByPetitionIdAndIsSupporting = originalCountArgs;
    });

    it('should display a content retrieval alert given no petition data', () => {
        const originalGetPetition = AppUtils.getPetition;
        AppUtils.getPetition = (id) => Promise.reject();

        const originalCountArgs = AppUtils.countArgsByPetitionIdAndIsSupporting;
        AppUtils.countArgsByPetitionIdAndIsSupporting = (_) => Promise.resolve(5);

        const props = {
            match: { params: { id: "testId" }},
            currentUser: { createdBy: { username: 'admin' }}
        };

        const component = shallow(createPetitionPage(props));
        component.setState({ isLoading: false });

        expect(toJson(component)).toMatchSnapshot();
        AppUtils.getPetition = originalGetPetition;
        AppUtils.countArgsByPetitionIdAndIsSupporting = originalCountArgs;
    });

});
