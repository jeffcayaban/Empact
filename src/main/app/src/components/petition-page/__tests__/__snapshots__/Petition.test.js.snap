// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Petition should display a content retrieval alert given no petition data 1`] = `
<div
  className="alertContainer"
>
  <ContentRetrievalAlert
    contentType="Petition"
  />
</div>
`;

exports[`Petition should render successfully 1`] = `
<div
  className="secondaryPageContainerStyle"
>
  <Grid
    bsClass="container"
    className="divNoPadding"
    componentClass="div"
    fluid={false}
    style={
      Object {
        "paddingTop": "15px",
      }
    }
  >
    <Col
      bsClass="col"
      className="noSidePadding"
      componentClass="div"
      lg={12}
      md={12}
    >
      <h1
        align="center"
      >
        TestTitle
      </h1>
      <div
        className="centerStyle"
        style={
          Object {
            "paddingTop": "10px",
          }
        }
      >
        <CreatedByOn
          content={
            Object {
              "action": "TestAction",
              "closingDateTime": "2019-02-15T13:04:27Z",
              "createdBy": Object {
                "authorities": Array [
                  "ROLE_ADMIN",
                ],
                "firstName": "Admin",
                "id": 1,
                "lastName": "Account",
                "username": "admin",
              },
              "creationDateTime": "2019-02-14T13:04:29Z",
              "goal": "TestGoal",
              "id": 1,
              "isAnonymous": false,
              "isWinning": true,
              "lastUpdatedDateTime": "2019-02-14T13:04:38Z",
              "situation": "TestSituation",
              "title": "TestTitle",
              "value": "TestValue",
            }
          }
          date="14-02-2019"
        />
        <LastUpdated
          creationDateTime="2019-02-14T13:04:29Z"
          lastUpdatedDateTime="2019-02-14T13:04:38Z"
        />
      </div>
    </Col>
  </Grid>
  <Grid
    bsClass="container"
    className="divNoPadding"
    componentClass="div"
    fluid={false}
    style={
      Object {
        "paddingTop": "40px",
      }
    }
  >
    <Col
      bsClass="col"
      className="noSidePadding"
      componentClass="div"
      lg={12}
      md={12}
    >
      <div
        className="spaceBetweenElements"
        style={
          Object {
            "paddingBottom": "15px",
          }
        }
      >
        <div
          className="contentIdNumber"
        >
          ID: 1
        </div>
      </div>
      <PetitionContent
        petition={
          Object {
            "action": "TestAction",
            "closingDateTime": "2019-02-15T13:04:27Z",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "Admin",
              "id": 1,
              "lastName": "Account",
              "username": "admin",
            },
            "creationDateTime": "2019-02-14T13:04:29Z",
            "goal": "TestGoal",
            "id": 1,
            "isAnonymous": false,
            "isWinning": true,
            "lastUpdatedDateTime": "2019-02-14T13:04:38Z",
            "situation": "TestSituation",
            "title": "TestTitle",
            "value": "TestValue",
          }
        }
        showCreatedByOnFooter={false}
      />
    </Col>
  </Grid>
  <CreateArgumentPanel
    closingDateTime="2019-02-15T13:04:27Z"
    hideCreateArgumentInvite={[Function]}
    isPetitionCreator={false}
    petitionId={1}
    showCreateArgument={true}
  />
  <Grid
    bsClass="container"
    className="divNoPadding"
    componentClass="div"
    fluid={false}
    style={
      Object {
        "paddingTop": "30px",
      }
    }
  >
    <Col
      bsClass="col"
      className="noSidePadding"
      componentClass="div"
      lg={12}
      md={12}
    >
      <ArgumentVisualisation
        contentId={1}
        isPetition={true}
      />
    </Col>
  </Grid>
  <Grid
    bsClass="container"
    className="divNoPadding"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={6}
        md={6}
      >
        <ArgumentPanel
          isCreator={false}
          isSupporting={true}
          parentContentType="Petition"
          parentCreatedById={1}
          subjectId={1}
        />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        lg={6}
        md={6}
      >
        <ArgumentPanel
          isCreator={false}
          isSupporting={false}
          parentContentType="Petition"
          parentCreatedById={1}
          subjectId={1}
        />
      </Col>
    </Row>
  </Grid>
</div>
`;
